### PR TITLE
Fix misspelling found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

File: dask/array/slicing.py [hunk 1]
```diff
@@ -1674,7 +1674,7 @@ def block_index_shape_from_1d_bool_index(dim, loc0, loc1):
 
     @functools.lru_cache
     def n_preceeding_from_1d_bool_index(dim, loc0):
```
Comment (DimitriPapadopoulos) [inline dask/array/slicing.py:1676]: The function name needs to be changed as well. Is it OK to change the name ad add a function with teh old name with a `FutureWarning` for deprecation?

File: dask/array/slicing.py [hunk 2]
```diff
@@ -1673,8 +1673,8 @@ def block_index_shape_from_1d_bool_index(dim, loc0, loc1):
         return np.sum(index[loc0:loc1])
 
     @functools.lru_cache
-    def n_preceeding_from_1d_bool_index(dim, loc0):
-        """Number of True index elements preceeding position loc0.
+    def n_preceding_from_1d_bool_index(dim, loc0):
```
Comment (DimitriPapadopoulos) [inline dask/array/slicing.py:1676]: Is it OK to change the function name? It's not part of the API as far as I can see — `setitem_array` is missing from the documentation.

Comment (phofl) [reply dask/array/slicing.py:1676]: I'd rather not change this, Dask and public API is a bit weird sometimes, even though I agree with you that nobody should be using it

Comment (DimitriPapadopoulos) [reply dask/array/slicing.py:1676]: Perhaps duplicate the function (it's very short) and add a deprecation notice to the existing one?

Comment (phofl) [reply dask/array/slicing.py:1676]: yep, you can just call the new function from the old one

Comment (DimitriPapadopoulos) [reply dask/array/slicing.py:1676]: I call the old deprecated function from the new one - I suspect that's what you meant.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**